rpcs3/Emu/NP/fb_helpers.cpp

```diff
@@ -67,7 +68,25 @@ namespace np
 		}
 	}
 
-	void np_handler::UserInfo2_to_SceNpUserInfo2(event_data& edata, const UserInfo2* user, SceNpUserInfo2* user_info, bool include_onlinename, bool include_avatarurl)
+	void UserInfo_to_SceNpUserInfo(const UserInfo* user, SceNpUserInfo* user_info)
+	{
+		if (user->npId())
+		{
+			std::memcpy(user_info->userId.handle.data, user->npId()->c_str(), std::min<usz>(16, user->npId()->size()));
+		}
+
+		if (user->onlineName())
+		{
+			std::memcpy(user_info->name.data, user->onlineName()->c_str(), std::min<usz>(48, user->onlineName()->size()));
+		}
+
+		if (user->avatarUrl())
+		{
+			std::memcpy(user_info->icon.data, user->avatarUrl()->c_str(), std::min<usz>(127, user->avatarUrl()->size()));
+		}
```

Comment:
you can do
`if (const auto npid = user->npId())` etc.
And then use the tmp var instead of getting the fb pointer each time